rust/template/src/ovsdb_api.rs

```diff
@@ -246,12 +255,14 @@ pub unsafe extern "C" fn ddlog_into_osvdb_delete_str(
     res
 }
 
-fn into_delete_str(table: *const c_char, rec: &Record) -> Result<CString, String> {
+fn into_delete_str(prog: &HDDlog, table: *const c_char, rec: &Record) -> Result<CString, String> {
```

Comment:
```suggestion
unsafe fn into_delete_str(prog: &HDDlog, table: *const c_char, rec: &Record) -> Result<CString, String> {
```

This function takes an arbitrary pointer and dereferences it, it's not safe